src/components/header/renderer.tsx

```diff
@@ -116,7 +120,11 @@ class Header extends React.PureComponent<Props, State> {
   }
 
   public onSwitchMode(option) {
-    option.value === Mode.Vega ? this.props.updateVegaSpec(stringify(this.props.vegaSpec)) : this.onSelectNewVegaLite();
+    if (option.value === Mode.Vega) {
+      this.props.updateVegaSpec(stringify(this.props.vegaSpec)) && this.onSelectNewVega();
```

Comment:
Should this also be an if or just two commands?

Reply:
We can put it in an if condition. The current state only shortens the code. Should I make another if?

Reply:
It's much clearer, yes.

Reply:
Oh! there was no need for if condition. The route will change every time. My bad

Reply:
I'm glad clarifying the code helped you as well.